src/scanner.rs

```diff
@@ -226,11 +274,17 @@ impl Scanner {
     /// Returns None if it's a hardlink of a file already seen.
     fn new_fileset(&mut self, path: &Path, metadata: &fs::Metadata) -> Option<RcFileSet> {
         let path: Box<Path> = path.into();
-        let device_inode = (metadata.dev(), metadata.ino());
+        let device_inode = (get_device(metadata), get_inode(metadata));
 
         match self.by_inode.entry(device_inode) {
```

Comment:
if Windows doesn't have inodes and hardlink counts, then the whole `by_inode` grouping doesn't make sense. Skip `self.by_inode` check entirely on Windows.